plugins/in_head/in_head.c

```diff
@@ -255,81 +255,29 @@ static int in_head_collect(struct flb_input_instance *i_ins,
 static int in_head_config_read(struct flb_in_head_config *ctx,
                                struct flb_input_instance *in)
 {
-    const char *filepath = NULL;
-    const char *pval = NULL;
-
-    /* filepath setting */
-    filepath = flb_input_get_property("file", in);
-    if (!filepath) {
+    int ret;
+    /* Load the config map */
+    ret = flb_input_config_map_set(in, (void *)ctx);
+    if (ret == -1) {
         return -1;
     }
-    ctx->filepath = filepath;
-
-    pval = flb_input_get_property("key", in);
-    if (pval) {
-        ctx->key      = pval;
-        ctx->key_len  = strlen(pval);
-    }
-    else {
-        ctx->key      = "head";
-        ctx->key_len  = 4;
-    }
 
-    /* buffer size setting */
-    pval = flb_input_get_property("buf_size", in);
-    if (pval != NULL && atoi(pval) > 0) {
-        ctx->buf_size = atoi(pval);
-    }
-    else {
-        ctx->buf_size = DEFAULT_BUF_SIZE;
-    }
 
-    /* interval settings */
-    pval = flb_input_get_property("interval_sec", in);
-    if (pval != NULL && atoi(pval) >= 0) {
-        ctx->interval_sec = atoi(pval);
-    }
-    else {
-        ctx->interval_sec = DEFAULT_INTERVAL_SEC;
-    }
+    ctx->key_len = strlen(ctx->key);
 
-    pval = flb_input_get_property("interval_nsec", in);
-    if (pval != NULL && atoi(pval) >= 0) {
-        ctx->interval_nsec = atoi(pval);
-    }
-    else {
-        ctx->interval_nsec = DEFAULT_INTERVAL_NSEC;
-    }
-
-    pval = flb_input_get_property("split_line", in);
-    if (pval != NULL && flb_utils_bool(pval)) {
-        ctx->split_line = FLB_TRUE;
+    // only set lines if not explicitly set
```

Comment:
Single line comment should be /* only set lines if not explicitly set */.
https://github.com/fluent/fluent-bit/blob/master/CONTRIBUTING.md#single-line-comment